[15.0][FIX] payment: Wait for payment methods to be created

Meisolsson opened the pull request:
Since other modules create account.payment.method in this migration we need to wait for them before finding all payment methods and and connecting them to the journal and acquirer.

Discussed in #4257

pedrobaeza commented:
Please change the commit name to include the module name.

https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#71commit-message

pedrobaeza commented on openupgrade_scripts/scripts/payment/15.0.2.0/post-migration.py:
```diff
@@ -68,7 +68,6 @@ def migrate(env, version):
     fill_payment_transaction_tokenize(env)
     fill_payment_transaction_partner_state_id(env)
     fill_payment_transaction_last_state_change(env)
-    create_account_payment_method_line(env)
```
Remove the method from here.

Meisolsson committed:
[FIX] payment: Delay payment method line migration

Since other modules create account.payment.method in this migration we
need to wait for them before finding all payment methods and and
connecting them to the journal and acquirer.

Meisolsson commented:
My bad, I was sure I had cleaned up the old migration and I've must have missed the note about adding the module name. Should be fixed now.
